### ISSUE TITLE
Invalid catalog name: 1046 No database selected

### ISSUE THREAD

**Issue by chrisidakwo** [spatie/laravel-multitenancy#253]
**Laravel Version: 8.44.0
Package Version: 1.8.1**

TL;DR: I'm having this error when I log in a user and try to redirect to another page. The error happens at the `AuthenticateSession` middleware while trying to resolve `$request->user()`. And yes, there's an active tenant accessible through `app('currentStore');`

**Error:** `Illuminate\Database\QueryException
SQLSTATE[3D000]: Invalid catalog name: 1046 No database selected (SQL: select * from `users` where `id` = e7beb12c-a2f8-4fea-9e5c-4de5744fcfbb limit 1) `

**My Setup:**
I'm using multiple databases and just getting started on the package. I've done all necessary setup including the 'tenant' middleware and listing tasks for database switching.

Now, I have a `/create-account` route that can be accessed by anyone and then `/email/verification` route that requires authentication. The `auth` middleware is called inside the controller's constructor.

<img width="929" alt="sc_web_routes" src="https://user-images.githubusercontent.com/29854560/120934361-fdaa0f80-c6f5-11eb-96da-8feed2d3102a.png">

Now, in my `register()` method within the `RegisterController`, I:
- create a new tenant
- create tenant database
- migrate tenant tables
- and create admin user for the tenant

All these work fine. The admin user is created in a `users` table within the tenant database.

However, I started having issues when I decided to log in the user and redirect to an email verification notice page.

Here's what the code looks like:

```php
// Setup new store
$store = $this->storeRepository->setupNewStore($storeData);

// Create first (admin) user for the store
$user = $this->userRepository->setupStoreAdmin($userData);

event(new Registered($user, $store));

// Set store as the active tenant
$store->makeCurrentTenant(); // Had to add this line to see if it fixes the issue

// Log in the user
Auth::guard()->login($user);

return redirect()->route('email.notice')->with([
	'status' => 'verification-sent',
	'email' => $user->email
]);
```

The `$store` variable is an object of the `Store` model class which extends the package's `Tenant` model. The `makeCurrentTenant()` method is an alias of `makeCurrent()`.

Now, the error stated above happens at the point of redirecting to the email notice view when it tries to resolve `$request->user()` in the `AuthenticateSession` middleware. But then, I have a current tenant setup with also sets a database on the config, so I'm not sure why the EloquentUserProvider can't get that database. I've tried to look at the framework source to see how the user provider gets the database but I guess I'm missing out something and need an extra eye. Here's some screenshot of the trace:

<img width="959" alt="sc_stack_trace_authenticate_middleware" src="https://user-images.githubusercontent.com/29854560/120966436-236aff00-c75e-11eb-92d1-7ead98b97c27.png">

<img width="957" alt="sc_stack_trace_session_guard" src="https://user-images.githubusercontent.com/29854560/120966413-1c43f100-c75e-11eb-9a6f-7c120206b35a.png">

**Comment by masterix21** [comment]
Hi @chrisidakwo. Where is your `users` table? In landlord or tenant database?

**Comment by chrisidakwo** [comment]
It's in the tenant database

**Comment by chrisidakwo** [comment]
I worked on a workaround for this. Doesn't directly fix the issue, but helps me get past it (for now) Here's what I did:

1.) I added an `X-DOMAIN` key with tenant uuid to the response cookies so it's available on the next request.
2.) I added a new middleware just above the `AuthenticateSession` middleware but below the `EncryptCookie` middleware so that I can receive the decrypted cookies.
3.) Still in the middleware, I retrieved the value of `X-DOMAIN` from the decrypted cookies and ran a DB query to get tenant record.
4.) On getting a valid record, I set it as the current tenant using `makeCurrent()`

And that's it.

**Comment by chrisidakwo** [comment]
After giving it some thought, I figured the error I'm getting is exactly what I should see (and please correct me if I'm wrong).

But I'm of the understanding that with a redirect, just like I have, a new request is made and the app goes through the whole framework's boot lifecycle again. Meaning that it'll lose any dynamic configuration that was set on the previous request. Hence the error.

Another way to fix this is to redirect the user to the tenant domain so that the `MultitenancyServiceProvider` can easily determine and set the current tenant. But then my requirement is different.

Thanks though!

**Comment by masterix21** [comment]
Please create a repository with your example code to test the error. Thanks.

I'm closing here because it isn't an issue related to the package, but feel free to reply.
